pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -219,6 +219,12 @@ func (e *eksOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 		if cluster.Status.ServiceAccountToken == "" || cluster.Status.APIEndpoint == "" {
 			cluster, err = e.generateAndSetServiceAccount(cluster)
 			if err != nil {
+				var statusErr error
+				cluster, statusErr = e.setFalse(cluster, apimgmtv3.ClusterConditionWaiting,
+					fmt.Sprintf("failed to generate service account, check networking settings: %v", err))
```

Comment:
```suggestion
					fmt.Sprintf("failed to generate service account: %v", err))
```
This should be more general, the call could be failing for any number of reasons, auth, networking outage. The error message coming back from `generateAndSetServiceAccount` should hopefully be useful enough to troubleshoot.